### ISSUE TITLE
Research Fine Tuning in General and Document findings and key advice

### ISSUE THREAD

**Issue by sara-stojkov** [sara-stojkov/AEOU-Garaza-Frontier-Hackathon#14]

**Comment by sara-stojkov** [comment]
Fine Tuning is used on general purpose models which are trained on various and great amounts of data. Fine tuning a model makes it better at a specific task, for example at sentiment analysis. 

Steps of fine tuning:
- pick a model
- prepare task specific data
- optionally modify model architecture, for example change output layer
- training:
- use a small learning rate to not change the model too much
- update model weights using backpropagation
- if working with a neural network, can change some or all layers

**Comment by sara-stojkov** [comment]
Useful links:
https://platform.openai.com/docs/guides/fine-tuning-best-practices
https://huggingface.co/docs/transformers/training
https://medium.com/@darrenoberst/6-tips-to-becoming-a-master-llm-fine-tuning-chef-143ad735354b
https://www.datacamp.com/tutorial/fine-tuning-large-language-models
